### PR TITLE
✨manager: add liveness/readiness probes

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:
Add liveness / readiness probes
similar did here: https://github.com/kubernetes-sigs/cluster-api/pull/2156

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
manager: Add liveness/readiness probes
```

/cc @vincepri

### CODE REVIEW

**Comment by randomvariable** [comment]
doesn't look like a flake. did fail on waiting on pod ready conditions

File: main.go [hunk 1]
```diff
@@ -183,6 +185,8 @@ func main() {
 		os.Exit(1)
 	}
 
+	setupChecks(mgr)
```
Comment (yastij) [inline main.go]: I'd expect this to be configurable for the users and to `9440` if no flag is passed

Comment (randomvariable) [reply main.go]: The flag was added in #764 ,but not wired up.
Need to add something along the lines of 
`HealthProbeBindAddress:  healthAddr`
to `ctrl.NewManager`

**Comment by randomvariable** [comment]
/lgtm

on account of wiring up the health-addr flag in #764

**Pull request merged**